src/rluni/robot/robot.py

```diff
@@ -0,0 +1,428 @@
+            # Start Loop Timer and increment loop iteration
+            loop_start_time = time.time()
+            self.itr += 1
+
+            # Debugging Data Example (add more data as needed)
+            tele_debug_data = td.DebugData()
+            tele_debug_data.add_data(example_debug_data=42)
+
+            # Sensor reading and fusion
+
+            imudata = td.IMUData(*self.imu.read_sensor_data(convert=True))
+
+            quaternion, internal_states, flags = self.sensor_fusion.update(
+                imudata.get_gyro(),
+                imudata.get_accel(),
+                mag_data=imudata.get_mag(),
+                delta_time=loop_period,
+            )
+            rigid_body_state = td.EulerAngles(*self.sensor_fusion.euler_angles, *self.sensor_fusion.euler_rates)
+            
+            # Update robot state and parameters
+            # if self.xmotor is not None:
+            #     await self.xmotor.update_state()
+
+            for motor in self.motors:
+                if motor is not None: 
+                    await motor.update_state()
+
+            # Control logic
+            # control_input = ControlInput(
+            #     pendulum_angle=rigid_body_state.y * DEG_TO_RAD,  # [radians] positive CCW
+            #     pendulum_vel=imudata.gyro_z * DEG_TO_RAD,  # [radians / s] positive CCW
+            #     wheel_vel=(
+            #         0
+            #         if self.xmotor is None
+            #         else -self.xmotor.state["VELOCITY"] * REV_TO_RAD
+            #     ),  # [radians / s] positive CCW
+            #     roll_torque=torque_request,  # [N * m] positive CCW
+            # )
+
+            control_input = ControlInput(
+                euler_angles_x_rads=rigid_body_state.x * DEG_TO_RAD,
+                euler_angles_y_rads=rigid_body_state.y * DEG_TO_RAD,
+                euler_angles_z_rads=rigid_body_state.z * DEG_TO_RAD,
+                euler_rates_x_rads_s=rigid_body_state.x_dot,
+                euler_rates_y_rads_s=rigid_body_state.y_dot,
+                euler_rates_z_rads_s=rigid_body_state.z_dot,
+                motor_speeds_pitch_rads_s=self.motors.pitch.state["VELOCITY"]*REV_TO_RAD,
+                motor_speeds_roll_rads_s=self.motors.roll.state["VELOCITY"]*REV_TO_RAD,
+                motor_speeds_yaw_rads_s=self.motors.yaw.state["VELOCITY"]*REV_TO_RAD,
+            )
+
+            # Change to negative convention due to motor
+            # torque_request = -self.controller.get_torque(
+            #     control_input, self.MAX_TORQUE - 0.001
+            # )
+            torques = self.controller.get_torques(
+                control_input, self.MAX_TORQUE - 0.001
+            )
+
+            ## DELAY UNTIL FIXED POINT ##
+            self.precise_delay_until(loop_start_time + loop_period * self.WRITE_DUTY)
+
+            # Apply control decision to robot actuators
+            # SET TORQUE
+            isCalibrating = self.itr < self.sensor_calibration_delay / self.LOOP_TIME
```

Comment:
We have lost the calibration sequence for the imu. The robot needs to be motionless for a calibration delay period to initate the sensor fusion. This could be an state in a state machine at some point. For now the conditional flag checks if the motors should be activated yet. The first self.sensor_calibration_delay seconds of time, the motors are suppressed

Reply:
```suggestion
                        isCalibrating = self.itr < self.sensor_calibration_delay / self.LOOP_TIME
            if self.motors_enabled and not isCalibrating:
                # Only set the torque if not in sensor fusion calibration mode
                await self.xmotor.set_torque(
                    torque=torque_request, max_torque=self.MAX_TORQUE
                )
````

This was the rwip code for reference